js/data/buffer.js

```diff
@@ -64,45 +70,53 @@ function Buffer(options) {
     this._createPushMethod();
 }
 
-// Start Buffer1 compatability code
-Buffer.prototype = {
-
-    get pos() { return this.length * this.itemSize; },
-    get index() { return this.length; },
+/**
+ * Bind this buffer to a WebGL context.
+ * @private
+ * @param gl The WebGL context
+ */
+Buffer.prototype.bind = function(gl) {
+    var type = gl[this.type];
 
-    get arrayType() { return this.type; },
+    if (!this.buffer) {
+        this.buffer = gl.createBuffer();
+        gl.bindBuffer(type, this.buffer);
+        gl.bufferData(type, this.arrayBuffer.slice(0, this.length * this.itemSize), gl.STATIC_DRAW);
```

Comment:
@jfirebaugh should we readd the `DataView` usage? I know browser sniffing is not nice, but if it means better performance for supporting browsers, maybe that's a fair tradeoff. I wouldn't like to use try/catch here too because it's on hot path and V8 doesn't optimize functions with try/catch.